test/printf.jl

```diff
@@ -81,19 +157,64 @@ end
 # chars
 @test (@sprintf "%c" 65) == "A"
 @test (@sprintf "%c" 'A') == "A"
+@test (@sprintf "%3c" 'A') == "  A"
+@test (@sprintf "%-3c" 'A') == "A  "
 @test (@sprintf "%c" 248) == "ø"
 @test (@sprintf "%c" 'ø') == "ø"
 
+# escape %
+@test (@sprintf "%%") == "%"
+@test (@sprintf "%%s") == "%s"
+@test_throws ArgumentError eval(:(@sprintf "%"))
+
+# argument count
+@test_throws ArgumentError eval(:(@sprintf "%s"))
+@test_throws ArgumentError eval(:(@sprintf "%s" "1" "2"))
+
+# unimplemented features
+@test_throws ErrorException eval(:(@sprintf "%n" "1"))
+@test_throws ErrorException eval(:(@sprintf "%'u" 1000))
+
+# type width specifier parsing (ignored)
+@test (@sprintf "%llf" 1.2) == "1.200000"
+@test (@sprintf "%Lf" 1.2) == "1.200000"
+@test (@sprintf "%hhu" 1) == "1"
+@test (@sprintf "%hu" 1) == "1"
+@test (@sprintf "%lu" 1) == "1"
+@test (@sprintf "%llu" 1) == "1"
+@test (@sprintf "%Lu" 1) == "1"
+@test (@sprintf "%zu" 1) == "1"
+@test (@sprintf "%ju" 1) == "1"
+@test (@sprintf "%tu" 1) == "1"
```

Comment:
These cases are all actually ignored by the current code. And if they weren't I'm not sure what the difference would be. They are supposed to describe the incoming type, in a dynamic language such information is unnecessary. This is just checking that they parse.